pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -538,6 +540,64 @@ public String convert(Object value, PinotDataType sourceType) {
     }
   },
 
+  JSON {
+    @Override
+    public int toInt(Object value) {
+      return Integer.parseInt(value.toString().trim());
+    }
+
+    @Override
+    public long toLong(Object value) {
+      return Long.parseLong(value.toString().trim());
+    }
+
+    @Override
+    public float toFloat(Object value) {
+      return Float.parseFloat(value.toString().trim());
```

Comment:
No need to trim, same for double (refer to the comments for STRING)

Reply:
Fixed.